webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -46,24 +48,53 @@ export default {
           variables,
         })
         if (!(markAsRead && markAsRead.read === true)) return
-        this.notifications = this.notifications.map(n => {
-          return n.from.id === markAsRead.from.id ? markAsRead : n
+        this.displayedNotifications = this.displayedNotifications.map(n => {
+          return this.equalNotification(n, markAsRead) ? markAsRead : n
         })
       } catch (err) {
-        throw new Error(err)
+        this.$toast.error(err.message)
       }
     },
+    equalNotification(a, b) {
+      return a.from.id === b.from.id && a.createdAt === b.createdAt && a.reason === b.reason
+    },
   },
   computed: {
-    totalNotifications() {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-26T07:11:19Z" title="Thursday, September 26th 2019, 9:11:19 am +02:00">Sep 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -29,13 +41,37 @@ export default {
   },
   data() {
     return {
+      displayedNotifications: [],
       notifications: [],
     }
   },
   props: {
     placement: { type: String },
   },
+  watch: {
+    notifications: {
+      immediate: true,
+      handler(lastQueriedNotifications) {
+        if (lastQueriedNotifications && lastQueriedNotifications.length > 0) {
```


Prefer guard clause over if statement

```suggestion
  if (!(lastQueriedNotifications && lastQueriedNotifications.length > 0)) return
```

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-26T08:16:41Z" title="Thursday, September 26th 2019, 10:16:41 am +02:00">Sep 26, 2019</time>_

---


Done

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-26T08:16:41Z" title="Thursday, September 26th 2019, 10:16:41 am +02:00">Sep 26, 2019</time>_

---


Done